### PR TITLE
fix(helm): fix race conditions in flag parsing

### PR DESCRIPTION
**Pull request by adamreese**
* fix a number of issues with flag parsing
* add support for `HELM_DEBUG`
* lazy expand flag default envars

### CODE REVIEW

**Review by technosophos — APPROVED** [review]
LGTM

File: cmd/helm/init.go [hunk 1]
```diff
@@ -54,8 +54,9 @@ To dump a manifest containing the Tiller deployment YAML, combine the
 `
 
 const (
-	stableRepository = "stable"
-	localRepository  = "local"
+	stableRepository         = "stable"
+	localRepository          = "local"
+	localRepositoryIndexFile = "index.yaml"
```
Comment (adamreese) [inline cmd/helm/init.go]: yes, just helped it home

**Comment by thomastaylor312** [comment]
@adamreese Before merging can you make sure this is labeled correctly and put into the proper milestone? I wasn't sure whether to put it in 2.4.3 or 2.5.0

**Comment by adamreese** [comment]
@technosophos one last look to get this merged?

File: cmd/helm/helm.go [hunk 2]
```diff
@@ -123,11 +120,12 @@ func newRootCmd(out io.Writer) *cobra.Command {
 		PersistentPreRun: func(cmd *cobra.Command, _ []string) {
 			initRootFlags(cmd)
 		},
-		PersistentPostRun: func(_ *cobra.Command, _ []string) {
+		PersistentPostRun: func(*cobra.Command, []string) {
 			teardown()
 		},
 	}
 	addRootFlags(cmd)
+	out := cmd.OutOrStdout()
```
Comment (technosophos) [inline cmd/helm/helm.go]: Nice. Never noticed that before.

**Review by technosophos — APPROVED** [review]
Okay -- re-read through the new commits, and all is looking good.

**Pull request merged**